components/ConfigurationError/ConfigurationError.tsx

```diff
@@ -0,0 +1,51 @@
+import { AplReadyResult } from "@saleor/app-sdk/APL";
+import { Alert } from "@saleor/macaw-ui";
+
+const vercelInfo = (
+  <div>
+    <Alert variant="warning" title="Vercel detected">
+      <p>
+        App detected Vercel hosting. Check if vercel is currently redeploying your app and try again
+        in few minutes
+      </p>
+    </Alert>
+  </div>
+);
+
+export function ConfigurationError({
+  isVercel,
+  appReady,
+}: {
+  isVercel: boolean;
+  appReady: AplReadyResult;
+}) {
+  return (
+    <div>
+      <h1>⚠️ Can&apos;t connect with the App API</h1>
+      {isVercel && vercelInfo}
+      {appReady.ready ? null : (
+        <div>
+          <p>Environment variables validation is incorrect</p>
```

Comment:
Errors can be related not only to variables?

```suggestion
          <p>APL healthcheck has failed:</p>
```